src/storage/recovery/recovery_manager.cpp

```diff
@@ -0,0 +1,934 @@
+        TERRIER_ASSERT(pr_map.find(catalog::RELOID_COL_OID) != pr_map.end(), "PR Map must contain class oid");
+        TERRIER_ASSERT(pr_map.find(catalog::RELNAME_COL_OID) != pr_map.end(), "PR Map must contain class name");
+        TERRIER_ASSERT(pr_map.find(catalog::RELKIND_COL_OID) != pr_map.end(), "PR Map must contain class kind");
+        auto next_class_oid = *(reinterpret_cast<uint32_t *>(
+            next_redo_record->Delta()->AccessWithNullCheck(pr_map[catalog::RELOID_COL_OID])));
+        auto next_class_kind UNUSED_ATTRIBUTE = *(reinterpret_cast<catalog::postgres::ClassKind *>(
+            next_redo_record->Delta()->AccessWithNullCheck(pr_map[catalog::RELKIND_COL_OID])));
+
+        if (class_oid == next_class_oid) {  // Condition 4: If the oid matches on the next record, this is a renaming
+          TERRIER_ASSERT(class_kind == catalog::postgres::ClassKind::REGULAR_TABLE && class_kind == next_class_kind,
+                         "We only allow renaming of tables");
+          // Step 4: Extract out the new name
+          VarlenEntry name_varlen = *(reinterpret_cast<VarlenEntry *>(
+              next_redo_record->Delta()->AccessWithNullCheck(pr_map[catalog::RELNAME_COL_OID])));
+          std::string name_string(name_varlen.StringView());
+
+          // Step 5: Rename the table
+          auto result UNUSED_ATTRIBUTE = GetDatabaseCatalog(txn, next_redo_record->GetDatabaseOid())
+                                             ->RenameTable(txn, catalog::table_oid_t(next_class_oid), name_string);
+          TERRIER_ASSERT(result, "Renaming should always succeed during replaying");
+
+          // Step 6: Update metadata and clean up additional record processed. We need to use the indexes on
+          // pg_class to find what tuple slot we just inserted into. We get the new tuple slot using the oid index.
+          auto pg_class_oid_index = GetDatabaseCatalog(txn, next_redo_record->GetDatabaseOid())->classes_oid_index_;
+          auto pr_init = pg_class_oid_index->GetProjectedRowInitializer();
+          buffer = common::AllocationUtil::AllocateAligned(pr_init.ProjectedRowSize());
+          pr = pr_init.InitializeRow(buffer);
+          *(reinterpret_cast<uint32_t *>(pr->AccessForceNotNull(0))) = next_class_oid;
+          std::vector<TupleSlot> tuple_slot_result;
+          pg_class_oid_index->ScanKey(*txn, *pr, &tuple_slot_result);
+          TERRIER_ASSERT(tuple_slot_result.size() == 1, "Index scan should only yield one result");
+          tuple_slot_map_[next_redo_record->GetTupleSlot()] = tuple_slot_result[0];
+          delete[] buffer;
+          tuple_slot_map_.erase(delete_record->GetTupleSlot());
+          delete[] reinterpret_cast<byte *>(next_redo_record);
+
+          return 1;  // We processed an additional record
+        }
+      }
+    }
+  }
+
+  // Step 3: If it was not a renaming, we call to the catalog to delete the class object
+  bool result UNUSED_ATTRIBUTE;
```

Comment:
It looks like it is truly unused.  Either use the result in a TERRIER_ASSERT (in which case we need the UNUSED to avoid problems in release) or remove this declaration and don't store the return value below.

Reply:
Its used on an assert below to ensure the delete succeeds